src/main/java/org/jenkinsci/test/acceptance/po/PluginManager.java

```diff
@@ -95,31 +95,34 @@ public PluginManager(Jenkins jenkins) {
      * Force update the plugin update center metadata.
      */
     public void checkForUpdates() {
-        mockUpdateCenter.ensureRunning(jenkins);
-        visit("index");
-        final String current = getCurrentUrl();
-        // The check now button is a form submit (POST) with a redirect to the same page only if the check is successful.
-        // We use the button itself to detect when the page has changed, which happens after the refresh has been done
-        // And we check for the presence of the button again
-        WebElement checkButton = find(by.link("Check now"));
-        checkButton.click();
-        // The wait criteria is: we have left the current page and returned to the same one
-        waitFor(checkButton).withTimeout(java.time.Duration.of(time.seconds(30), ChronoUnit.MILLIS)).until(webElement -> {
-            try {
-                // We interact with the element just to detect if it is stale
-                webElement.findElement(by.id("it does not matter"));
-            } catch(StaleElementReferenceException e) {
-                // with this exception we know we've left the original page
-                // we look for an element in the page to check for success
-                if (current.equals(getCurrentUrl())) {
-                    return true;
+        boolean isRunning = mockUpdateCenter.ensureRunning(jenkins);
```

Comment:
`isRunning` is a misleading variable name.  the method will return `false` even when the `mockUpdateCenter` is correctly injected and running.  It only returns `true` the first time the mockUpdateCenter is injected.

The code is kind of right - but I think we need to maybe make `checkForUpdates` always check for updates, but possible all calling code should be changed to not call this before attempting to install plugins - and that the constructor is this should maybe not call it but call say another method.

Reply:
What are your thoughts if we split `checkForUpdates` into two parts, and call `ensureRunning` where always, but `clickCheckNow` before plugin installation if list of plugins was not refreshed before? see [cf72035](https://github.com/jenkinsci/acceptance-test-harness/pull/913/commits/cf72035f7920c813736e64bdf805dc92a74a699d)